### PR TITLE
adds ER update via service and token

### PR DESCRIPTION
**Pull request by andrewpmartinez**
Allows ER lists to be obtain in context of the current identity via `GET /edge/client/v1/service/{id}/edge-routers`

### CODE REVIEW

File: client.yml [hunk 1]
```diff
@@ -2863,6 +2863,130 @@ paths:
...
       in: path
       required: true
+  /services/{id}/edge-routers:
+    get:
+      security:
+      - ztSession: []
+      - oauth2:
+        - openid
+      description: |
+        Retrieves a list of edge router resources that a specific service can dial or bind.
```
Comment (ekoby) [inline client.yml:2873]: that sounds weird to me
how about `list of edge router that can be used to dial/bind given service`?

File: source/client/services.yml [hunk 2]
```diff
@@ -76,13 +76,62 @@ paths:
         '429':
           $ref: '../shared/standard-responses.yml#/responses/rateLimitedResponse'
 
+  services-id-edge-routers:
+    parameters:
+      - $ref: '../shared/parameters.yml#/id'
+    get:
+      summary: List of edge routers that a service can be dialed/bound for the current identity
```
Comment (ekoby) [inline source/client/services.yml:83]: again, sounds awkward
`... that current identity can use to dial/bind given service`?

File: client.yml [hunk 3]
```diff
@@ -2863,6 +2863,130 @@ paths:
...
+      tags:
+      - Service
+      summary: List of edge routers that a service can be dialed/bound for the current
+        identity
```
Comment (ekoby) [inline client.yml:2877]: I guess same as in the source

File: client.yml (hunk 1 above)
Comment (dovholuknf) [reply client.yml:2873]: "Retrieves the list of edge routers authorized to dial or bind the specified service" ?

File: client.yml [hunk 4]
```diff
@@ -2863,6 +2863,130 @@ paths:
...
+    get:
+      security:
+      - ztSession: []
+      - oauth2:
+        - openid
+      description: |
+        Retrieves a list of edge router resources that a specific service can dial or bind.
+      tags:
+      - Service
+      summary: List of edge routers that a service can be dialed/bound for the current
```
Comment (dovholuknf) [inline client.yml:2876]: same comment here as above wrt description

**Review by ekoby — APPROVED** [review]

File: client.yml (hunk 1 above)
Comment (ekoby) [reply client.yml:2873]: sounds wrong because ERs are not dialing or binding, they are used to dial/bind by an identity

File: client.yml [hunk 5]
```diff
@@ -2863,6 +2863,130 @@ paths:
...
+        name: limit
+        in: query
+      - type: integer
+        name: offset
+        in: query
+      - type: string
+        name: filter
+        in: query
+      - type: string
+        description: an optional JWT token to validate, if provided a not authorized
```
Comment (dovholuknf) [inline client.yml:2890]: Also reads weird to me...  is this supplemental to the header? does it als0 validate the token? 

how about:
"an optional JWT token use to authenticate the request. If provided, the token needs to be valid else a not authorized response is returned"

File: client.yml [hunk 6]
```diff
@@ -2863,6 +2863,130 @@ paths:
...
+          response is returned if the token is not valid
+        name: session-token
+        in: header
+      responses:
+        "200":
+          description: A list of dial/bind edge routers for the requesting identity
```
Comment (dovholuknf) [inline client.yml:2896]: --> "A list of edge routers suitable for dialing or binding the specified service by the authenticated identity" ?

File: client.yml (hunk 1 above)
Comment (dovholuknf) [reply client.yml:2873]: yeah. probably can just remove the dial/bind thing entirely? "Retrieves the list of edge routers usable by the specified service" ?

**Pull request merged**